app/models/agents/telegram_agent.rb

```diff
@@ -0,0 +1,93 @@
+require 'httmultiparty'
+require 'open-uri'
+require 'tempfile'
+
+module Agents
+  class TelegramAgent < Agent
+    include HTTMultiParty
+    base_uri 'https://api.telegram.org/'
+
+    cannot_be_scheduled!
+    cannot_create_events!
+    no_bulk_receive!
+
+    description <<-MD
+      The Telegram Agent receives and collects events and sends them via [Telegram](https://telegram.org/).
+
+      It is assumed that events have either a `text`, `photo`, `audio`, `document` or `video` key. You can use the EventFormattingAgent if your event does not provide these keys.
+
+      The value of `text` key is sent as a plain text message.
+      The value of `photo`, `audio`, `document` and `video` keys should be an url which contents are sent to you according to the type.
+
+      **Setup**
+
+      1. obtain an `auth_token` by [creating a new bot](https://telegram.me/botfather).
+      2. [send a private message to your bot](https://telegram.me/YourHuginnBot)
```

Comment:
Since it'll be Markdown, people may not see the URL, so I'd suggest

"Send a private message to your bot by visiting https://telegram.me/YourHuginnBot"